app/models/invoice.js

```diff
@@ -108,14 +109,23 @@ export default AbstractModel.extend(DateFormat, NumberFormat, {
   patientResponsibility: Ember.computed.sum('patientResponsibilityTotals'),
 
   paymentAmountChanged: function() {
-    var payments = this.get('payments'),
-      paidTotal = payments.reduce(function(previousValue, payment) {
+    var payments = this.get('payments');
```

Comment:
I think this code block could be simplified to just keep track of if the payments are valid and then set the status to paid if remainingBalance <=0 and hasValidPayments === true.  There really isn't a need to do anything if those conditions are untrue.  There shouldn't be a need to reset the status back to the status it is already set to.